### PR TITLE
Installation: Add flatpak_installation_get_min_free_space_bytes

### PR DESCRIPTION
**Pull request by uajain**
A convenience wrapper around ostree_repo_get_min_free_space_bytes().
This can be used to query the min-free-space-* config option of the
repo by clients like gnome-software.

https://phabricator.endlessm.com/T23937

### CODE REVIEW

File: common/flatpak-installation.c [hunk 1]
```diff
@@ -1519,6 +1519,40 @@ flatpak_installation_get_config (FlatpakInstallation *self,
...
 
+/**
+ * flatpak_installation_get_min_free_space_bytes:
+ * @self: a #FlatpakInstallation
+ * @out_bytes: (out): Location to store the result
+ * @error: Return location for a #GError
+ *
+ * A convenience wrapper around ostree_repo_get_min_free_space_bytes().
+ * This can be used to query the min-free-space-* config option of the
+ * repo by clients like gnome-software.
```
Comment (wjt) [inline common/flatpak-installation.c]: I think @matthiasclasen's point at https://github.com/flatpak/flatpak/commit/04b9e5af0afbba0cf5fd42e9b4b5c921f951795e#r31038814 still stands:

> The documentation should tell me what the function does and what i would want to use it for. What are apps supposed to do with this number ?

Comment (wjt) [reply common/flatpak-installation.c]: (There's a lot of rationale in the commit message, but that's not what API users will see.)

Comment (matthiasclasen) [reply common/flatpak-installation.c]: One question I still have is: why is this value stored in ostree config - is ostree using it for its own purposes ?
If so, why do clients have to do their own manual computation ?

File: common/flatpak-installation.c [hunk 2]
```diff
@@ -1519,6 +1519,40 @@ flatpak_installation_get_config (FlatpakInstallation *self,
...
+ *
+ * Returns: %TRUE on success, or %FALSE on error.
+ * Since: 1.1.0
+ */
```
Comment (matthiasclasen) [inline common/flatpak-installation.c]: In other places, I have just used

> Since:

We should be consistent with this

 1.1

**Comment by matthiasclasen** [comment]
Also, not the greatest form to refer in the commit message to a phabricator instance that doesn't let me register.

File: common/flatpak-installation.c (hunk 1 above)
Comment (matthiasclasen) [reply common/flatpak-installation.c]: Can we change the documentation to something like:

Returns the min-free-space config value from the OSTree repository of this installation.

Applications can use this value, together with information about the available disk space and
the size of pending updates/installs, to estimate whether a pull operation will fail due to running
out of disk space.

**Comment by matthiasclasen** [comment]
It would be neat to add a test in tests/testlibrary.c for this new api, so we can keep up the coverage, in particular for public library api.

**Comment by matthiasclasen** [comment]
@rh-atomic-bot: r+

**Comment by uajain** [comment]
> broken_heart Test failed - [status-papr](https://s3.amazonaws.com/aos-ci/ghprb/flatpak/flatpak/da80f665c294bb8620dbf495b3bf402f9e62cc66.1540518053129908224/index.html)

Looks like OSTree version 2018.9 has not hit the package-management repos yet.

**Comment by barthalion** [comment]
@rh-atomic-bot: r+

**Comment by TingPing** [comment]
@rh-atomic-bot: r+

**Comment by TingPing** [comment]
@rh-atomic-bot retry

**Comment by matthiasclasen** [comment]
https://bodhi.fedoraproject.org/updates/FEDORA-2018-f15ba17ffa

**Comment by matthiasclasen** [comment]
bot, retest this please

**Comment by uajain** [comment]
On my local system:
```
[uajain@localhost ~]$ sudo dnf --repo updates-testing install ostree
Last metadata expiration check: 1:20:45 ago on Thu 01 Nov 2018 02:01:41 PM IST.
Dependencies resolved.
================================================================================
 Package           Arch        Version               Repository            Size
================================================================================
Installing:
 ostree            x86_64      2018.9-1.fc28         updates-testing      226 k
Upgrading:
 ostree-devel      x86_64      2018.9-1.fc28         updates-testing      157 k
 ostree-libs       x86_64      2018.9-1.fc28         updates-testing      382 k

Transaction Summary
================================================================================
Install  1 Package
Upgrade  2 Packages

Total download size: 765 k
Is this ok [y/N]: y
```

**Comment by matthiasclasen** [comment]
it should be in updates by now, I think. I got it the karma it needed yesterday

**Comment by ramcq** [comment]
bot, retest this please

**Comment by matthiasclasen** [comment]
bot, retest this, _please_

**Comment by matthiasclasen** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**